app/scripts/modules/core/src/pipeline/config/stages/wait/WaitStageConfig.tsx

```diff
@@ -0,0 +1,88 @@
+import { IStageConfigProps } from 'core/pipeline/config/stages/core/IStageConfigProps';
+import * as React from 'react';
+
+import { StageConfigField } from '../core/stageConfigField/StageConfigField';
+import { SpelNumberInput } from 'core/widgets/spelText/SpelNumberInput';
+import { IStage } from 'core/domain';
+
+export interface IWaitStageConfigState {
+  enableCustomSkipWaitText: boolean;
+  waitTime: number | string;
+  skipWaitText: string;
+}
+
+export const DEFAULT_SKIP_WAIT_TEXT = 'The pipeline will proceed immediately, marking this stage completed.';
+
+export class WaitStageConfig extends React.Component<IStageConfigProps, IWaitStageConfigState> {
+  constructor(props: IStageConfigProps) {
+    super(props);
+    this.state = this.getState(props.stage);
+  }
+
+  private getState(stage: IStage): IWaitStageConfigState {
+    if (!stage.waitTime) {
+      stage.waitTime = 30;
+    }
+    return {
+      enableCustomSkipWaitText: !!stage.skipWaitText,
+      waitTime: stage.waitTime,
+      skipWaitText: stage.skipWaitText,
+    };
+  }
+
+  private updateWaitTime = (waitTime: number | string) => {
+    this.props.stage.waitTime = waitTime;
+    this.setState({ waitTime });
+    this.props.stageFieldUpdated.next();
```

Comment:
since no data is being provided to the subject, could we just use a callback?

Reply:
yes!!!